### DISCUSSION TITLE
SetShaderParameter should throw a warning/error if Paramater doesn't exist

### DISCUSSION THREAD

**Discussion by louiidev** [godotengine/godot-proposals#9360] [Engine Core]
Would be helpful for debugging issues with shaders if the engine threw a warning/error when trying to set a shader parameter that doesn't exist on the shader

**Comment by AThousandShips** [comment]
This might not be entirely simple, setting the parameters happens asynchronously to avoid synchronizing with rendering and this means it's set via caches that are then updated the next time a transfer is done, this could be handled with a check but that would mean adding more data to handle that, this would be more expensive in cases when you don't need this check when you've made sure the parameter does exist

**Reply by louiidev** [reply]
Yeah fair enough, if it’s at a cost of performance it’s probably not worth it

**Reply by timothyqiu** [reply]
I guess this could be included in debug build and editor build (when not running as editor).

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-proposals/issues/1494.

This makes a lot of sense (also for AnimationTree parameters), but we'll need to add `set_or_null()` alternatives in case you don't want an error message when setting a shader parameter that may not exist.
